src/python/WMCore/Services/pycurl_manager.py

```diff
@@ -47,20 +47,54 @@
 
 # system modules
 import json
+import gzip
 import logging
 import os
 import re
 import subprocess
 import sys
 import pycurl
 from io import BytesIO
+import io
 import http.client
 from urllib.parse import urlencode
+import traceback
 
 from Utils.Utilities import encodeUnicodeToBytes, decodeBytesToUnicode
 from Utils.PortForward import portForward, PortForward
 
 
+def decompress(body, headers):
+    """
+    Helper function to decompress given body if HTTP headers contains gzip encoding
+    :param body: bytes
+    :param headers: dict
+    :return: decode body
+    """
+    encoding = ""
+    for header, value in headers.items():
+        if header.lower() == 'content-encoding' and 'gzip' in value.lower():
+            encoding = 'gzip'
+            break
+    if encoding != 'gzip':
+        return body
+
+    try:
+        zbuf = io.BytesIO()
+        zbuf.write(body)
+        zbuf.seek(0)
+        zfile = gzip.GzipFile(mode='rb', fileobj=zbuf)
+        data = zfile.read()
+        zfile.close()
+        return data
+    except Exception as exc:
+        logger = logging.getLogger()
+        msg = "while processing decompress function with headers\n:%s" % headers
+        msg += "we were unable to decompress gzip content:\n%"
+        msg += str(exc)
+        logger.exception(msg)
+        return body
```

Comment:
This here I'd say is a good strategy to return the body (same for the `return body` above) in case the server has set the header inappropriately and the contents are actually not zipped, but then wouldn't it be worth checking the returned value bellow and if this was the exit point from the `decompress` method, then we also retry the regular: 

```
data = curl.bbuf.getvalue().decode('utf-8')
```

I am talking about lines 511-513 from the current PR. And I can see the type of behavior I am talking about is implemented at lines 393,394 inside the `multirequest` method.